www-servers/nginx/nginx-1.19.8.recipe

```diff
@@ -0,0 +1,120 @@
+SUMMARY="Robust, small and high performance http and reverse proxy server"
+DESCRIPTION="nginx [engine x] is an HTTP and reverse proxy server, a mail proxy \
+server, and a generic TCP proxy server, originally written by Igor Sysoev. \
+For a long time, it has been running on many heavily loaded Russian sites \
+including Yandex, Mail.Ru, VK, and Rambler. According to Netcraft, nginx \
+served or proxied 21.89% busiest sites in June 2015."
+HOMEPAGE="http://nginx.org/"
+COPYRIGHT="2002-2019 Igor Sysoev
+	2011-2019 Nginx, Inc."
+# "it's complicated", cf. http://packages.gentoo.org/package/www-servers/nginx
+LICENSE="BSD (2-clause)"
+REVISION="1"
+SOURCE_URI="https://nginx.org/download/nginx-$portVersion.tar.gz"
+CHECKSUM_SHA256="308919b1a1359315a8066578472f998f14cb32af8de605a3743acca834348b05"
+PATCHES="nginx-$portVersion.patchset"
+ADDITIONAL_FILES="
+	userlaunch.in
+	setup-nginx.sh"
+
+ARCHITECTURES="?x86_gcc2 ?x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+GLOBAL_WRITABLE_FILES="
+	settings/nginx directory keep-old
+	"
+
+POST_INSTALL_SCRIPTS="
+	$relativePostInstallDir/setup-nginx.sh
+	"
+
+#FIXME: on ./auto/configure:
+#grep: /etc/group: No such file or directory
+PROVIDES="
+	nginx$secondaryArchSuffix = $portVersion
+	cmd:nginx$secondaryArchSuffix = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libpcre$secondaryArchSuffix
+	lib:libssl$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libpcre$secondaryArchSuffix
+	devel:libssl$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:awk
+	cmd:find
+	cmd:gcc$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:make
+	cmd:ranlib
+	cmd:sed
+	"
+
+BUILD()
+{
+	./configure --prefix=$prefix \
+		--sbin-path=$binDir/nginx \
+		--doc-prefix=$docDir \
```

Comment:
With this the default html pages are still installed in $docdir (index.html and 50.html)
Maybe better move them to $localStateDir here also:
```diff
- 		--doc-prefix=$docDir \
+ 		--doc-prefix=$localStateDir/www/nginx \
```
**EDIT** like this you won't have to create the directory in INSTALL but you should add "var/www/nginx" as directory keep old in GLOBAL_WRITABLE_FILES

Reply:
I did it this way so that the files that come with nginx source are always up-to-date. It's unlikely that anyone installing nginx is going to leave the default pages and will likely delete them. Removing the symlinks ensures the source files are still available.

Reply:
Didn't though about that, good point :)